publisher/internal/analytics/ethereum/operations.go

```diff
@@ -167,39 +169,53 @@ func (rem Removal) Extract(eLog EventLog) (Operation, error) {
 		},
 	}
 	remPosition = rem.includeTokenPrices(remPosition)  // 6) Getting token prices
-	remPosition = calculatePosition(eLog, remPosition) // 7) Save Liquidity Entry and Liquidity Pool
+	remPosition = calculatePosition(burn, remPosition) // 7) Save Liquidity Entry and Liquidity Pool
 	rem.Position = remPosition
 	return rem, nil
 }
 
-func (add Addition) Extract(eLog EventLog) (Operation, error) {
-	if !isUniswapPositionsNFT(eLog.Data) {
+func (add Addition) Extract(mint EventLog) (Operation, error) {
+	if !isUniswapPositionsNFT(mint.Data) {
 		return Addition{}, fmt.Errorf("not Uniswap Positions NFT.\n")
 	}
-	// 2) Mint event is found
-	var addPosition Position
-	txEventsFromCache, _ := add.cache.Get(eLog.TransactionHash)
+
+	txEventsFromCache, _ := add.cache.Get(mint.TransactionHash)
+	addPosition := Position{
+		Address: mint.Address,
+		TxHash:  mint.TransactionHash,
+	}
+
 	for _, evLog := range txEventsFromCache.([]EventLog) { // Go through all events of this transaction
-		if reflect.DeepEqual(evLog, eLog) { // This allows to correctly process many Mint events in one transaction
-			break
-		}
-		if isAvoidEvent(evLog) {
-			addPosition = Position{} // Reset gathered records
-			continue
-		}
-		if isTransferEvent(evLog) { // 3) Searching for relevant "Transfer" event(s)
-			addPosition = add.handleLiquidityTransfer(evLog, addPosition) // 4) Decoding / expanding "Transfer" events
+		if isTransferEvent(evLog) && !isUniswapPositionsNFT(evLog.Address) {
+			addPosition = add.handleLiquidityTransfer(mint, evLog, addPosition)
 		}
 	}
+
 	if isEitherTokenUnknown(addPosition) {
-		addPosition = add.checkAndUpdateMissingToken(eLog, addPosition) // 5) Adding missing token if only 1 token transfer was made
+		addPosition = add.checkAndUpdateMissingToken(mint, addPosition) // 5) Adding missing token if only 1 token transfer was made
 	}
+	add.savePool(addPosition)
+
 	addPosition = add.includeTokenPrices(addPosition)  // 6) Getting token prices
-	addPosition = calculatePosition(eLog, addPosition) // 7) Save Liquidity Entry and Liquidity Pool
+	addPosition = calculatePosition(mint, addPosition) // 7) Save Liquidity Entry and Liquidity Pool
 	add.Position = addPosition
+
 	return add, nil
 }
 
+func (add Addition) savePool(addPos Position) {
+	if isEitherTokenAmountIsZero(addPos) || isEitherTokenUnknown(addPos) {
+		return
+	}
+	// In this case both tokens were transferred to LP and their order is correct
+	var newLiqPoll repository.Pool
+	newLiqPoll.Address = addPos.Address
+	newLiqPoll.Token0Address = addPos.Token0.Address
+	newLiqPoll.Token1Address = addPos.Token1.Address
+	add.db.SavePool(newLiqPoll)
```

Comment:
It would be nice not to ignore the error here. Perhaps add an error return type to `savePool` and simply return it, then handle somewhere, where `savePool` is used.

Reply:
Added the missed error with handling